libs/withdraws/src/lib/use-withdraw-asset.tsx

```diff
@@ -0,0 +1,58 @@
+import { captureException } from '@sentry/react';
+import type { Asset } from '@vegaprotocol/react-helpers';
+import { addDecimal } from '@vegaprotocol/react-helpers';
+import { AccountType } from '@vegaprotocol/types';
+import BigNumber from 'bignumber.js';
+import { useCallback, useEffect } from 'react';
+import type { Account } from './types';
+import { useGetWithdrawDelay } from './use-get-withdraw-delay';
+import { useGetWithdrawThreshold } from './use-get-withdraw-threshold';
+import { useWithdrawStore } from './withdraw-store';
+
+export const useWithdrawAsset = (
+  assets: Asset[],
+  accounts: Account[],
+  assetId?: string
+) => {
+  const { asset, balance, min, threshold, delay, update } = useWithdrawStore();
+  const getThreshold = useGetWithdrawThreshold();
+  const getDelay = useGetWithdrawDelay();
+
+  // Every time an asset is selected we need to find the corresponding
+  // account, balance, min viable amount and delay threshold
+  const handleSelectAsset = useCallback(
+    async (id: string) => {
+      const asset = assets.find((a) => a.id === id);
+      const account = accounts.find(
+        (a) =>
+          a.type === AccountType.ACCOUNT_TYPE_GENERAL &&
+          a.asset.id === asset?.id
+      );
+      const balance =
+        account && asset
+          ? new BigNumber(addDecimal(account.balance, asset.decimals))
+          : new BigNumber(0);
+      const min = asset
+        ? new BigNumber(addDecimal('1', asset.decimals))
+        : new BigNumber(0);
+      // Query collateral bridge for threshold for selected asset
+      // and subsequent delay if withdrawal amount is larger than it
+      let threshold;
+      let delay;
+      try {
+        const result = await Promise.all([getThreshold(asset), getDelay()]);
+        threshold = result[0];
+        delay = result[1];
+      } catch (err) {
+        captureException(err);
+      }
+      update({ asset, balance, min, threshold, delay });
```

Comment:
question (non-blocking): because threshold is of type `BigNumber` it will get updated to `undefined` when failed - isn't it breaking anything?

Reply:
I haven't changed this bit too much, but I agree, this seems a bit off, so I am moving update inside the try.

Reply:
updated